src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,138 @@
+package com.commercetools.sync.commons.utils;
+
+import io.sphere.sdk.client.BlockingSphereClient;
+import io.sphere.sdk.client.QueueSphereClientDecorator;
+import io.sphere.sdk.client.RetrySphereClientDecorator;
+import io.sphere.sdk.client.SphereAccessTokenSupplier;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.client.SphereClientConfig;
+import io.sphere.sdk.http.AsyncHttpClientAdapter;
+import io.sphere.sdk.http.HttpClient;
+import io.sphere.sdk.retry.RetryAction;
+import io.sphere.sdk.retry.RetryContext;
+import io.sphere.sdk.retry.RetryPredicate;
+import io.sphere.sdk.retry.RetryRule;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+
+import javax.annotation.Nonnull;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+public class RetryableSphereClientWithExponentialBackoff {
```

Comment:
you don't have any public method for this class. Also, the class might be final and has a private constructor because you will use the factory method `of` for object construction.